docs/source/serving.rst

```diff
@@ -1,12 +1,468 @@
+
+    pip install vllm --pre --extra-index-url https://wheels.vllm.ai/nightly
+    pip install --pre torchao --index-url https://download.pytorch.org/whl/nightly/cu126
+
+.. code-block:: bash
+
+    # Server
+    vllm serve pytorch/Phi-4-mini-instruct-float8dq --tokenizer microsoft/Phi-4-mini-instruct -O3
+
+    # Client
+    curl http://localhost:8000/v1/chat/completions -H "Content-Type: application/json" -d '{
+    "model": "appy1234/Phi-4-mini-instruct-float8dq",
+    "messages": [
+        {"role": "user", "content": "Give me a short introduction to large language models."}
+    ],
+    "temperature": 0.6,
+    "top_p": 0.95,
+    "top_k": 20,
+    "max_tokens": 32768
+    }'
+
+.. note::
+    For more information on vLLM Integration, please refer to the detailed guide :ref:`torchao_vllm_integration`.
+
+
+Inference with Transformers
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Install the required packages:
+
+.. code-block:: bash
+
+    pip install git+https://github.com/huggingface/transformers@main
+    pip install torchao
+    pip install torch
+    pip install accelerate
+
+.. code-block:: python
+
+    import torch
+    from transformers import AutoModelForCausalLM, AutoTokenizer, pipeline
+
+    torch.random.manual_seed(0)
+
+    model_path = "pytorch/Phi-4-mini-instruct-float8dq"
+
+    model = AutoModelForCausalLM.from_pretrained(
+        model_path,
+        device_map="auto",
+        torch_dtype="auto",
+        trust_remote_code=True,
+    )
+    tokenizer = AutoTokenizer.from_pretrained(model_path)
+
+    messages = [
+        {"role": "system", "content": "You are a helpful AI assistant."},
+        {"role": "user", "content": "Can you provide ways to eat combinations of bananas and dragonfruits?"},
+        {"role": "assistant", "content": "Sure! Here are some ways to eat bananas and dragonfruits together: 1. Banana and dragonfruit smoothie: Blend bananas and dragonfruits together with some milk and honey. 2. Banana and dragonfruit salad: Mix sliced bananas and dragonfruits together with some lemon juice and honey."},
+        {"role": "user", "content": "What about solving an 2x + 3 = 7 equation?"},
+    ]
+
+    pipe = pipeline(
+        "text-generation",
+        model=model,
+        tokenizer=tokenizer,
+    )
+
+    generation_args = {
+        "max_new_tokens": 500,
+        "return_full_text": False,
+        "temperature": 0.0,
+        "do_sample": False,
+    }
+
+    output = pipe(messages, **generation_args)
+    print(output[0]['generated_text'])
+
+Mobile Deployment with ExecuTorch
+--------------------------------
+
+ExecuTorch enables on-device inference using torchao's mobile-optimized quantization schemes. The 8da4w (8-bit dynamic activation, 4-bit weight) configuration is specifically designed for mobile deployment.
+
+Step 1: Untie Embedding Weights
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+We want to quantize the embedding and lm_head differently. Since those layers are tied, we first need to untie the model:
```

Comment:
Is this step necessary actually? I don't think I had to do any of this for Llama models for example. Can you share the source for this?